x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_selector/index.tsx

```diff
@@ -187,15 +198,15 @@ export const ConversationSelector: React.FC<Props> = React.memo(
     useEvent('keydown', onKeyDown);
 
     useEffect(() => {
-      setSelectedOptions(conversationOptions.filter((c) => c.label === selectedConversationId));
+      setSelectedOptions(conversationOptions.filter((c) => c.id === selectedConversationId));
     }, [conversationOptions, selectedConversationId]);
 
     const renderOption: (
       option: ConversationSelectorOption,
       searchValue: string,
       OPTION_CONTENT_CLASSNAME: string
     ) => React.ReactNode = (option, searchValue, contentClassName) => {
-      const { label, value } = option;
+      const { label, value, id } = option;
```

Comment:
Since conversations are now unique by ID and not name (which was previously acting as the ID), you can have what appears to be duplicate conversations, for which the `EuiComboBox` will render all instances as 'selected'.

Happy with deferring this by preventing the UI from creating dupes for now (and fix once @patrykkopycinski pushes the new conversation selection UI), or we could sort by ID for all dupes and update the label to be the ole fan favorite `Conversation (1)`, `Conversation (2)`. Or something else, whatever your preference is 🙂 

<img width="1405" alt="image" src="https://github.com/elastic/kibana/assets/2946766/b8a35ca7-654c-4dc2-b78d-c92509811c09">

Reply:
Great catch! 
BTW: Are you able to create the duplicates with current UI? I can only do that with API.

Reply:
In the UI. I was able to type into the conversation selector and press enter to make a new conversation with the same name.

Reply:
I decided just to disable creation conversation with the same title:
<img width="841" alt="Screenshot 2024-01-31 at 8 37 47 PM" src="https://github.com/elastic/kibana/assets/55110838/28adf235-0f23-4e9f-86d0-32a1caab36b7">